src/_guides/language/effective-dart/design.md

```diff
@@ -880,26 +880,47 @@ var buffer = new StringBuffer()
 
 ## Type annotations
 
-In Dart 1.x, adding static types to your variables is optional.
+A "type annotation" is a place in your code where you write down the static type
+of a variable, field, function return type, parameter, etc. Type annotations are
+how you tell the type checker your constraints on the kinds of values that flow
+into different parts of your code.
 
-{% include optional-types-2.0.html %}
+In most places, Dart allows you to omit a type annotation and infers a type for
+you based on the nearby context (or defaults to the `dynamic` type). Inference
```

Comment:
There are two cases where inference can give you `dynamic` as a result:

```dart
// 1. Successfully infer dynamic.
var list = <dynamic>["stuff"];
var i = list[0]; // <-- Infer "dynamic" for i based on type of list.

// 2. Fail to infer, default to dynamic.
var a; // <-- Fail to infer a and use dynamic.
```

It's important to keep these cases distinct because they represent different things and have different user impact. Also, the second case is an error when it happens at the top-level or when you have `--no-implicit-dynamic` enabled, but the first case isn't.